src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -498,6 +498,26 @@ private void validatePluginArguments() {
         // finally, give each plugin a chance to trim down any unseen instances from it's own list
         pluginDescriptors.entrySet().forEach(e -> e.getValue().validateArguments());
     }
+    /**
+     * Check the provided value against any range constraints specified in the Argument annotation
+     * for the corresponding field. Throw an exception if limits are violated.
+     *
+     * - Only checks numeric types (int, double, etc.)
+     */
+    private void checkArgumentRange(final ArgumentDefinition argumentDefinition, final Object argumentValue) {
+        // Only validate numeric types
+        if ( ! (argumentValue instanceof Number) ) {
+            return;
+        }
+        final double argumentDoubleValue = ((Number)argumentValue).doubleValue();
+
+        // Check hard limits first, if specified
+        if ((argumentDefinition.minValue != Double.NEGATIVE_INFINITY && argumentDoubleValue < argumentDefinition.minValue)
+            || (argumentDefinition.maxValue != Double.POSITIVE_INFINITY && argumentDoubleValue > argumentDefinition.maxValue)) {
+            throw new CommandLineException.OutOfRangeArgumentValue(argumentDefinition.fullName, argumentDefinition.minValue, argumentDefinition.maxValue, argumentDoubleValue);
+        }
+    }
```

Comment:
I rebased, added a logger in the class and log a warning if the value is out of the soft boundaries.